BACKLOG.md

```diff
@@ -0,0 +1,38 @@
+BACKLOG
+=======
+
+General refactoring
+-------------------
+There are 3 representation of a tree.
+
+Remove `LTree`. `LTree` is currently not exported from `muste-lib`, it
+is however used internally. E.g. in `linearizeTree` which is used in
+the ajax-backend.
+
```

Comment:
There is also LinToken and Linearization (in Muste.hs) - should be merged into one.

Reply:
Which functions fit best where? 

- linearization functions can be moved to a new module (e.g. Muste.Linearize): LinToken, Linearization, linearizeTree (are defined in Muste.hs)

- the Ajax backend should only refer to functions exported by Muste.hs: now Ajax.hs imports Muste.Tree, Main-CGI.hs imports Muste.Grammar, and Database.hs and Protocol.hs imports both Muste.Tree, Muste.Grammar and PGF

- perhaps Database.hs should be moved to the muste-backend?? Also, Database.hs contains explicit example trees

- is there a need for Muste.Grammar.Internal? and Muste.Tree.Internal? 

- only Muste.Grammar (or Muste.Grammar.Internal) should use the PGF module -- everything that the other modules need should be exported by Muste.Grammar

- Muste.Feat currently defines the types Grammar, Rule, FunType -- this should be done by Muste.Grammar (and the functions getAllRules, getRuleType)

- Muste.Feat currently defines TTree -- this should be in Muste.Tree instead

- functions in Muste.Prune that should be in other files:
   - getFunctions, sameRoot, getMetas, treeDiff, treeIsRecursive, ... (move to Muste.Tree)
   - editDistance, isSubList, areDisjoint, noDuplicates, ... (move to Muste.Common)